### ISSUE TITLE
[check-send-result] not correctly checking instruction after `send`

### ISSUE THREAD

**Issue by StErMi** [protofire/solhint#295]
I've setup solhint and I get some error that I don't know how I should fix (usually with normal lint they also provide examples of good and bad behaviour)
for example:

```
bool success = payable(walletAddress).send(moneyAmount);
require(success, "Failed to send the moneyAmount back");
```

it says Linter: `Check result of "send" call [check-send-result]`

but as you can see I'm already testing it. How should that code be done? I think that (I'm not an expert to create the rules but I'm used to eslint and tslint usage) that it should check the result of `send` is used after that into a `require`

**Comment by dbale-altoros** [comment]
@StErMi 
Fisrt of all. Sorry for the late late late response to this.
Second. You're right. This is a false positive
The way to not get that error will be:

`require(payable(walletAddress).send(moneyAmount), "Failed to send the moneyAmount back");`
or with Custom Errors
`error FailedToSendMoneyAmountBack();
  function newGreeting(address walletAddress, uint256 moneyAmount) public {
          if(!payable(walletAddress).send(moneyAmount)) {
            revert FailedToSendMoneyAmountBack();
        }
  }`

I will update the docs to specify this issue

Thanks a lot for posting
There will be a new release available today and we will be updating solhint on regular basis with a dedicated team to it

**Comment by dbale-altoros** [comment]
readme updated
